docs pass & omnigraph overview (#2142)
namehash/ensnode: docs/ensnode.io/src/content/docs/docs/integrate/why-ensnode.mdx

title: Why ENSNode?
description: Why ENSNode is the best way to build ENSv2 apps.
---

import { LinkCard, CardGrid } from "@astrojs/starlight/components";
import LostENSNamesImage from "@workspace/docs/ensnode.io/src/components/atoms/images/astro/LostENSNamesImage.astro";
import ENSAdminImage from "@workspace/docs/ensnode.io/src/components/atoms/images/astro/ENSAdminImage.astro";

ENSNode is the full-stack ENSv2 development platform — a single, unified API over **both ENSv1 and ENSv2**, with first-class multichain support and the integration surfaces (React, TypeScript, GraphQL, Postgres) you need to ship.

## Built for ENSv2 from day one

> [ENSv2](https://roadmap.ens.domains/roadmap/) is a fundamental change to ENS's onchain data model. The legacy ENS Subgraph wasn't designed for it, and it never will be — single-chain, no resolution, no understanding of the new Namegraph. Apps that stay on the Subgraph will be left behind the moment ENSv2 launches.

Reviewer: ```suggestion
[ENSv2](https://ens.domains/ensv2) is a fundamental change to ENS's onchain data model. The legacy ENS Subgraph wasn't designed for it, and it never will be — single-chain, no resolution, no understanding of the new ENSv2 Namegraph. Apps that stay on the Subgraph will be left behind the moment ENSv2 launches.
```

ENSNode is built around ENSv2 from the ground up. The same query works against an ENSv1 name today and an ENSv2 name the moment it goes live — with **zero downtime or code changes** in your app.

<LinkCard
  title="ENSv2 Readiness"
  description="What you need to do to prepare your app for ENSv2."
  href="/docs/integrate/ensv2-readiness"
/>

## One unified API over ENSv1 + ENSv2

> The [ENS Omnigraph](/docs/integrate/omnigraph) is a single GraphQL API that returns a polymorphic, unified view of every ENS Domain — regardless of whether it lives in the ENSv1 nametree, the ENSv2 Namegraph, on mainnet, or on an L2 subregistry.

Reviewer: ```suggestion
The [ENS Omnigraph](/docs/integrate/omnigraph) is a single GraphQL API that returns a polymorphic, unified view of every ENS Domain — regardless of whether it lives in an ENSv1 Nametree, the ENSv2 Namegraph, on mainnet, or on an L2 subregistry such as base.eth subnames on Base.
```

![ENS Omnigraph diagram](/ens-omnigraph-diagram.png)

> ENSv1 and ENSv2 coexist after the ENSv2 launch. The Omnigraph hides that split: your code asks for `domain(by: { name: "vitalik.eth" })` and gets a typed result, regardless of whether the underlying domain is on ENSv1 or ENSv2.

Reviewer: ```suggestion
ENSv1 and ENSv2 coexist after the ENSv2 launch. The Omnigraph simplifies that split: your code asks for `domain(by: { name: "vitalik.eth" })` and gets a typed result, regardless of whether the underlying domain is on ENSv1 or ENSv2.
```

## Multichain by default

> ENSNode indexes the full ENS namespace across every chain that matters, all from one query:

Reviewer: ```suggestion
ENSNode provides the world's best indexing coverage for ENS, spanning every chain that matters, including:
```

- **Mainnet** — the canonical ENS root and `.eth` registrations
- **Basenames** (`.base.eth`) on Base
- **Lineanames** (`.linea.eth`) on Linea
- **3DNS** names (`.box`, tokenized DNS) on Optimism and Base

> No more wiring up a per-chain Subgraph, reconciling overlapping namespaces, or writing CCIP-Read logic in your app — ENSNode handles forward and reverse resolution, including offchain gateways, server-side.

Reviewer: ```suggestion
No more wiring up a per-chain Subgraph, reconciling overlapping ENS Nametrees, or writing CCIP-Read logic in your app — ENSNode handles forward and reverse resolution, including offchain gateways, server-side.
```
